src/main/java/org/motechproject/prevac/service/impl/PrimeVaccinationScheduleServiceImpl.java

```diff
@@ -67,19 +71,19 @@ public PrimeVaccinationScheduleDto createOrUpdateWithDto(PrimeVaccinationSchedul
             return new PrimeVaccinationScheduleDto(updateVisitWithDto(primeVisit, screeningVisit, dto));
```

Comment:
you should not return the PrimeVaccinationScheduleDto here, because number of patients will non be checked, add else to this if and put all code below in it (except the clinic fetch and number of patients check)